build/cluster.tf

```diff
@@ -13,160 +13,138 @@
+      "${var.cluster["name"]}", "${var.cluster["project"]}",
+      "${var.cluster["machineType"]}", "${var.cluster["initialNodeCount"]}",
+    "${var.cluster["zone"]}")}"}"
+  }
 }
 
 resource "google_container_cluster" "primary" {
-  name     = "${lookup(var.cluster, "name")}"
-  location     = "${lookup(var.cluster, "zone")}"
-  project  = "${lookup(var.cluster, "project")}"
+  name     = "${var.cluster["name"]}"
+  location = "${var.cluster["zone"]}"
+  project  = "${var.cluster["project"]}"
   provider = "google-beta"
 
-  # Setting an empty username and password explicitly disables basic auth
-  # TODO(roberthbailey): Remove the entire master_auth block when switching to 1.12.
-  master_auth {
-    username = "${local.username}"
-    password = "${var.password}"
+  node_pool {
+    name       = "default"
+    node_count = "${var.cluster["initialNodeCount"]}"
+
+    node_config {
+      machine_type = "${var.cluster["machineType"]}"
 
-    client_certificate_config {
-      issue_client_certificate = false
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring",
+        "https://www.googleapis.com/auth/service.management.readonly",
+        "https://www.googleapis.com/auth/servicecontrol",
+        "https://www.googleapis.com/auth/trace.append",
+      ]
+
+      tags = ["game-server"]
     }
   }
-  enable_legacy_abac = "${lookup(var.cluster, "legacyAbac")}"
-  node_pool = [
-    {
-      name = "default"
-      node_count = "${lookup(var.cluster, "initialNodeCount")}"
-      node_config = {
-        machine_type = "${lookup(var.cluster, "machineType")}"
-        oauth_scopes = [
-          "https://www.googleapis.com/auth/devstorage.read_only",
-          "https://www.googleapis.com/auth/logging.write",
-          "https://www.googleapis.com/auth/monitoring",
-          "https://www.googleapis.com/auth/service.management.readonly",
-          "https://www.googleapis.com/auth/servicecontrol",
-          "https://www.googleapis.com/auth/trace.append",
-        ]
-
-        tags = ["game-server"]
-        timeouts = {
-          create = "30m"
-          update = "40m"
-        }
+
+  node_pool {
+    name       = "agones-system"
+    node_count = 1
+
+    node_config {
+      machine_type = "n1-standard-4"
+
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring",
+        "https://www.googleapis.com/auth/service.management.readonly",
+        "https://www.googleapis.com/auth/servicecontrol",
+        "https://www.googleapis.com/auth/trace.append",
+      ]
+
+      labels = {
+        "stable.agones.dev/agones-system" = "true"
```

Comment:
The label (and taint) should just be `agones.dev/agones-system` now.

Reply:
thanks for an update